File: 1394-find-lucky-integer-in-an-array/1394-find-lucky-integer-in-an-array.java
class Solution {
    public int findLucky(int[] arr) {
        HashMap<Integer, Integer> map = new HashMap<Integer, Integer>();
        int res = -1;
        
        for (int i : arr){
            map.put(i, map.getOrDefault(i,0) + 1);
        }
        
        for (Map.Entry<Integer, Integer> entry : map.entrySet()){
            if (entry.getKey() == entry.getValue()){
                res = Math.max(res, entry.getKey());
            }
        }
        return res;
    }
}

File: 1394-find-lucky-integer-in-an-array/1394-find-lucky-integer-in-an-array.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        d = {}
        m = -1
        for i in arr:
            d[i] = d.get(i, 0) + 1
        
        for j in d.keys():
            if d[j] == j:
                m = max(m, j)
            
        return m